templates/tweets/scripts.html

```diff
@@ -0,0 +1,52 @@
+<script type="text/javascript">
+  const getCookie = name => {
+    if (document.cookie && document.cookie !== "") {
+      for (const cookie of document.cookie.split(";")) {
+        const [key, value] = cookie.trim().split("=");
+        if (key === name) {
+          return decodeURIComponent(value);
+        }
+      }
+    }
+  };
+  const csrftoken = getCookie('csrftoken');
+
+
+  const changeStyles = jsonResponse => {
+    const selector = document.getElementsByName(jsonResponse.tweet_id);
+    if (jsonResponse.liked) {
+      console.log("liked")
+      selector.innerHTML = "<i class='fas fa-lg fa-heart like-red'></i>";
+    } else {
+      console.log("not liked")
+      selector.innerHTML = "<i class='far fa-lg fa-heart'></i>";
+    }
```

Comment:
いいねしたときに、いいねの色が変わらないのは、１６行目で取ってきている値がNodeListという配列だからです。
ですので、NodeListに.innerHTMLメソッドをつけたところで、NodeList直下にはinnerHTMLメソッドがないため反映できていない状態になっています。

では、どうすれば反映するのか。innerHTMLメソッドがある場所に行けばいいのです。
現状のselectorの値をconsole.logで見てみると、以下のようになっています。
```
NodeList [button#like]
0: button#like
innerHTML: "<i class='fas fa-lg fa-heart like-red'></i>"
length: 1
[[Prototype]]: NodeList
```
ここからわかるのは、innerHTMLを直接変えにいくには、NodeListの0番目の値を参照すればいいことがわかります。ですので、
```suggestion
const selector = document.getElementsByName(jsonResponse.tweet_id)[0];
```
とすれば、innerHTMLに参照できるようになり、いいねの色が変わると思います。

参考として、[こちら](https://qiita.com/takeshisakuma/items/9e0c3b9800c307740593)から、どのようなdocumentオブジェクトのメソッドを使ったらどのような形で取ってくるのか(NodeList形式なのかHTMLCollection形式なのか)を確認してみてください。
ちなみに、NodeList直下で使えるメソッドは[こちら](https://developer.mozilla.org/ja/docs/Web/API/NodeList#%E3%83%A1%E3%82%BD%E3%83%83%E3%83%89)から確認できます。

Reply:
修正しました。